### PR TITLE
refactor: extract class for body parsing

### LINKED ISSUE #50: ApiRepoDeleteSlice.java:33-37: Remove code duplication....
The puzzle `321-1d437500` from #321 has to be resolved: 

https://github.com/artipie/management-api/blob/ffe7cf4e00381c3782e3b31ea60b047d4a3f8f36/src/main/java/com/artipie/management/api/ApiRepoDeleteSlice.java#L33-L37

The puzzle was created by Alexander on 29-Nov-21. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by genryxy**
Closes #50 
Extracted class for parsing body content for receiving some content

### CODE REVIEW

**Comment by genryxy** [comment]
@g4s8 could you please check

**Comment by genryxy** [comment]
@g4s8 could you please take a look

**Comment by genryxy** [comment]
@g4s8 please check

File: src/main/java/com/artipie/management/api/ApiRepoPostRtSlice.java [hunk 1]
```diff
@@ -65,23 +64,26 @@ public Response response(
                 .thenApply(
                     form -> {
                         final Response res;
-                        final String action = ApiRepoPostRtSlice.value(form, "action");
-                        if (Action.UPDATE.value().equals(action)) {
+                        final ValueFromBody vals = new ValueFromBody(form);
+                        final Optional<String> meth = vals.byName("method");
```
Comment (g4s8) [inline src/main/java/com/artipie/management/api/ApiRepoPostRtSlice.java:68]: SHouldn't it be `action`? https://github.com/artipie/management-api/blob/master/src/main/resources/dashboard/repo.hbs#L46

File: src/main/java/com/artipie/management/api/ApiRepoPostRtSlice.java [hunk 2]
```diff
@@ -65,23 +64,26 @@ public Response response(
...
+                        if (meth.isPresent() && Action.UPDATE.value().equals(meth.get())) {
                             res = new ApiRepoUpdateSlice(this.configfile)
                                 .response(
-                                    line,
-                                    headers,
-                                    new Content.From(form.getBytes(StandardCharsets.US_ASCII))
+                                    line, headers,
+                                    new Content.From(
+                                        vals.payload().getBytes(StandardCharsets.US_ASCII)
```
Comment (g4s8) [inline src/main/java/com/artipie/management/api/ApiRepoPostRtSlice.java:74]: Are you sure we always use ASCII for configurayion? E.g. can we name some repository or user with unicode symbols? If we can, then we need to decode it with UTF-8.

File: src/main/java/com/artipie/management/misc/ValueFromBody.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        this.cpayload = payload;
+        this.charset = charset;
+    }
+
+    /**
+     * Ctor.
+     * @param payload Information which was passed in the body
+     */
+    public ValueFromBody(final String payload) {
+        this(payload, StandardCharsets.US_ASCII);
```
Comment (g4s8) [inline src/main/java/com/artipie/management/misc/ValueFromBody.java:44]: Same question here: why we can't give unicode name to some form fields?

**Review by g4s8 — CHANGES_REQUESTED** [review]
Sorry for delay (each time I tried to find this code https://github.com/artipie/management-api/blob/master/src/main/resources/dashboard/repo.hbs#L46). Please check several comments/questions

File: src/main/java/com/artipie/management/api/ApiRepoPostRtSlice.java (hunk 1 above)
Comment (genryxy) [reply src/main/java/com/artipie/management/api/ApiRepoPostRtSlice.java]: I missed this one. I stashed some changes and after that apply this changes but got conflict and after fixing this one missed this name

**Comment by genryxy** [comment]
@g4s8 thanks, corrected. As I understand there are not (in code and docs) any restrictions about using unicode symbols in repo configuration as [here](https://github.com/artipie/artipie/blob/master/src/main/java/com/artipie/RepoConfig.java#L189) it is read in `UTF-8` therefore replaced `ASCII` with `UTF-8`

**Review by g4s8 — APPROVED** [review]
@genryxy thanks

**Pull request merged**